libraries/node-core-library/src/JsonFile.ts

```diff
@@ -139,6 +146,21 @@ export class JsonFile {
     return jsonObject;
   }
 
+  /**
+   * Validates a JSON file's contents against its schema.
+   */
+  public static validate(
+    jsonContents: string,
+    jsonFilename: string,
```

Comment:
The `jsonFilename` here is the `filenameForErrors` parameter. We definitely need to document that clearly.

Reply:
Again with two adjacent `string` parameters it seems easy to mix them up.

Since `jsonSchema` is more important information, and more important information should come first, perhaps the order could be:
```
validate(json, jsonSchema, filenameForErrors, options);
```

Reply:
On the other hand, is this:
```
JsonFile.validateString(json, jsonSchema, filenameForErrors, options);
```
...actually superior to this:
```
const jsonObject: JsonObject = JsonFile.parseString(someString, filenameForErrors);
myJsonSchema.validateObject(jsonObject, filenameForErrors, options);
```
The second one is more typing, but it seems easier to understand. And how often do we write this code anyway?  Is the savings shorthand really worth the hassle of having yet another API?